graphql-language/src/ast.rs

```diff
@@ -59,20 +60,18 @@ impl_node_for! { FloatValue }
 ///   - `""`
 ///   - `"` StringCharacter+ `"`
 pub struct StringValue<'a> {
-  pub loc: Option<Location<'a>>,
-  pub value: String
+  pub loc: Location<'a>,
+  pub value: &'a str
 }
 
 impl_node_for! { StringValue }
 
 /// Document : Definition+
 pub struct Document<'a> {
-  pub loc: Option<Location<'a>>,
+  pub source: Source<'a>,
   pub definitions: Vec<Definition<'a>>
 }
 
-impl_node_for! { Document }
```

Comment:
`Document` is a node, it must `impl Node`. Also, what happens when there is whitespace padding a source document? Location would not be the entire source.